### DISCUSSION TITLE
What programming languages and frameworks are supported by GitHub Copilot?

### DISCUSSION THREAD

**Discussion by pkofficial46** [community/community#132097] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Help !

**Comment by nadeeshaofficial1** [comment, marked as answer]
GitHub Copilot supports numerous programming languages and frameworks, including JavaScript, Python, Java, Ruby, Go, PHP, C++, C#, and Swift. It also works with frameworks like React, Angular, Vue.js, Node.js, Django, Flask, and Ruby on Rails. This AI tool enhances productivity by providing code suggestions, auto-completion, and generating code snippets across these languages and frameworks.

**Reply by renan-alm** [reply]
Sharing a weird question from MS Accreditation program. As far as I knew there wasn't something called "supported languages". All official languages would work with various quality levels. Am I wrong? 

<img width="1000" height="479" alt="image" src="https://github.com/user-attachments/assets/877f38b0-6145-4cb1-8a1e-c2266e2480bc" />

**Comment by NielsLiisberg** [comment]
Is there a way to participate in the project and apply more subtle languages like RPGLE ?

**Comment by xuhdev** [comment]
The official supported list is buried under [GitHub language support](https://docs.github.com/en/get-started/learning-about-github/github-language-support#core-languages-supported-by-github-features). There's a table listing supported languages of GitHub copilot. At the time of writing (Dec 2024), it supports: C, C++, C#, Go, Java, JavaScript, Kotlin, PHP, Python, Ruby, Rust, Scala, Swift, Typescript.

However, other languages may also work:

> Most GitHub features work regardless of which languages your code is written in.

**Comment by krahulsahu** [comment]
GitHub Copilot supports most popular programming languages and frameworks, including:

Languages: Java, JavaScript, TypeScript, Python, C#, C++, Go, PHP, Ruby, Kotlin, Swift, Rust, SQL, and more.
Frameworks & Libraries: Spring Boot, React, Angular, Vue.js, Node.js, Express.js, Next.js, Django, Flask, ASP.NET, Laravel, Ruby on Rails, and many others.

Copilot works best with widely used languages and frameworks that have large amounts of public code and documentation available.
